common/app/rendering/core/JavascriptProps.scala

```diff
@@ -0,0 +1,21 @@
+package rendering.core
+
+import conf.Configuration
+import play.api.libs.json.{JsValue, Json, Writes}
+
+case class JavascriptPropsConfig(
+  bundleUrl: String = conf.Static("javascripts/ui.bundle.browser.js"),
+  polyfillioUrl: String =
+    if (conf.switches.Switches.PolyfillIO.isSwitchedOn) common.Assets.js.polyfillioUrl
+    else conf.Static("javascripts/vendor/polyfillio.fallback.js"),
+  beaconUrl: String = Configuration.debug.beaconUrl
+)
```

Comment:
I would recommend not to pass default values here (instead pass those values in the `default` method below), so one cannot create an instance of `JavascriptPropsConfig` without knowing what goes in it.